src/storage/src/hummock/sstable/forward_sstable_iterator.rs

```diff
@@ -247,24 +120,44 @@ impl SstableIterator {
         // do cooperative scheduling.
         tokio::task::consume_budget().await;
 
+        let mut hit_cache = false;
         if idx >= self.sst.value().block_count() {
             self.block_iter = None;
-        } else {
+            return Ok(());
+        } else if let Some(preload_stream) = self.preload_stream.as_mut() && preload_stream.next_block_index() <= idx {
```

Comment:
Do we need some kind of retry here if the preload_stream returns an error? Previously we only use streaming read in compaction and compaction is running in full speed without backpressure so the stream will be polled frequently. However, now streaming read can be used in streaming/batch query, where backpressure can happen. When backpressure happens, the stream may not be polled in time. Not sure whether this can be a problem.

Reply:
Good catch, I will let it retry with some times.